actions/get_dependencies/action.yaml

```diff
@@ -9,6 +9,17 @@ runs:
     - name: Give Permission
       run: chmod +x ${{github.action_path}}/get_dependencies.sh 
       shell: bash 
+    - name: cache-libint
+        id: cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            libint-2.6.0/
+            install/
+          key: libint-gcc${{ matrix.gcc_version }}
```

Comment:
This does not seem like an appropriate place to put this logic since 90% of invocations of this action will not build libint. Can we factor out this logic and place it somewhere closer to the actual building of libint? This may require a special dispatch for libint, say to a special action.